controllers/drplacementcontrol_controller.go

```diff
@@ -1502,22 +1502,20 @@ func (r *DRPlacementControlReconciler) getManagedClusterResource(
 		return errorswrapper.Wrap(err, "getManagedClusterResource failed")
 	}
 
-	// get query results
-	recentConditions := rmnutil.GetMostRecentConditions(mcv.Status.Conditions)
+	r.Log.Info(fmt.Sprintf("MCV condtions: %v", mcv.Status.Conditions))
 
-	r.Log.Info(fmt.Sprintf("MCV recent condtions: %v", recentConditions))
 	// want single recent Condition with correct Type; otherwise: bad path
-	switch len(recentConditions) {
+	switch len(mcv.Status.Conditions) {
```

Comment:
Doesn't this statement rarely produces the expected results.  IOW, doesn't this input take you to the default case even though it is a good input?
```
status:
  conditions:
  - lastTransitionTime: "2021-08-03T22:03:55Z"
    message: Apply manifest work complete
    reason: AppliedManifestWorkComplete
    status: "True"
    type: Applied
  - lastTransitionTime: "2021-08-03T22:03:56Z"
    message: All resources are available
    reason: ResourcesAvailable
    status: "True"
    type: Available
```

Reply:
I checked the following sources to confirm the number of conditions:
1) view [doc](https://github.com/open-cluster-management/multicloud-operators-foundation/blob/main/docs/view/managedClusterView.md)
2) Code:
- [api](https://github.com/open-cluster-management/multicloud-operators-foundation/blob/dda6b80090951ca7abcc9bc8aca86d13c9202b26/pkg/apis/view/v1beta1/view_types.go#L92-L104)
- [reconcile code](https://github.com/open-cluster-management/multicloud-operators-foundation/blob/dda6b80090951ca7abcc9bc8aca86d13c9202b26/pkg/klusterlet/view/view_controller.go#L80-L156)

The above suggests that MCV will only have at most 1 condition, and hence the code above.

Am I looking at the wrong place? I checked my test cluster as well and did not see "ResourceAvailable" condition.

Checked the forked [repo](https://github.com/tjanssen3/multicloud-operators-foundation/tree/main/v2) as well for completeness and it remains as above.

Reply:
You are right, only one condition.